### ISSUE TITLE
About evaluation on maestro set using dcase2024.py 

### ISSUE THREAD

**Issue by frednam93** [merlresearch/sebbs#1]
Hi, 

Thanks for sharing your great work! I read your paper and it was very interesting. I am trying to run dcase2024.py on the score files I obtained using dcase2024 baseline (modified but did not change the test much), and I have some questions.

1. The function "eval_desed" makes error because some "clip_id"s in "ground_truth" is not in "scores". I think I have the all validation set (1168 files) and I don't know why some clip_ids are missing in scores. Do you have any clue?

2. I noticed that in "write_sed_scores", key error occurs due to desed files missing maestro class keys and vice versa. Did you expect all the files to include both desed and maestro classes as keys?

3. I noticed that the dcase2024 baseline outputs scores after merging the files divided into 10-seconds-long while the maestro ground truth tsv file corresponds to divided 10-seconds-long files. So this raises error when I put the directory for maestro ground truth tsv for "test_ground_truth_maestro".  Did you intend the users to merge the ground truth by themselves or save tsv files for each 10-seconds-long files?

4. The "keys" in "eval_maestro" uses "classes_labels_maestro_real_eval" while the dcase2024 baseline uses "classes_labels_maestro_real", so this also raised key error. Is there any specific reason you chose eval class dict?

5. I managed to merger the dcase2024 baseline maestro ground truth and run eval_maestro, but then the mpauc before and after csebbs are exactly the same. Is this supposed to be so?

Thanks in advance!

Regards,
Hyeonuk Nam

**Comment by JanekEbb** [comment]
Hi,

thanks for your interest in our work :) You are raising great points here, which I am happy to help resolving them.

1. I guess you are using the scores from baseline test? Here the issue is that the validation set (ground truth file) includes files which actually have no active events, which are ignored in the baseline test and hence no files get written for them. You should use "eval_from_checkpoint" instead (after [this PR](https://github.com/DCASE-REPO/DESED_task/pull/106) was merged), which writes outputs for all files.

2. Same as above. When you are using scores written during desed test, only scores for desed classes are included in the files. With eval_from_checkpoint, however, outputs for all classes get written.

3. Oh yea, the script actually does the merging of the clip scores and expects the maestro ground truth tsv to provide long file ground truth. So you can either write such file using the ground truth merge functions from the baseline recipe or concatenate corresponding annotations from DESED_task/data/dcase/dataset/MAESTRO_original/maestro_dev. Sorry for the inconvenience here!

4. While there are a total of 17 maestro classes in "classes_labels_maestro_real", only 11 of them ("classes_labels_maestro_real_eval") are actually used for evaluation (also in the baseline).

5. Yes that's how it should be. We did have tuned cSEBBs for maestro classes at some point, but it didn't make a big difference given that for maestro classes scores are averaged over 1 sec segments anyway (so a precise localization of event boundaries is less important). So we decided to not have all that extra code in there and apply cSEBBs only to DESED classes and, to reduce output file size, already average maestro scores over 1 sec segments, which'll be done in evaluation anyways. So if you are not interested in the maestro performance when running the script, you don't even have to provide maestro ground truth tsv as it is optional.

Let me know if something remains unclear or you have more questions!

**Comment by frednam93** [comment]
Thanks for quick reply! I just realized that I was supposed to use evaluation=True setting for this, but thanks anyway! :D

**Comment by frednam93** [comment]
Oops sorry! I just found out that even if I use score obtained by evaluation=True, I get the error in the first question: key error as clip_id from ground truth is not in the scores. Could you check on this?

**Comment by frednam93** [comment]
Hi, I found out that unprocessed mpauc score reported by your code quite higher than the scrore given by dcase2024 baseline (~0.04). Is this supposed to be this way as well?

**Comment by JanekEbb** [comment]
No that wasn't the case for us. But note that baseline reports mpauc for postprocessed scores only which in our case didn't really differ from mpauc of unprocessed scores though. In contrast to last year, however, this year's baseline's unprocessed scores apparently do perform significantly worse than postprocessed scores in terms of PSDS. So maybe you changed something that improved unprocessed scores and now they perform better in terms of mpauc than postprocessed scores?

**Comment by JanekEbb** [comment]
Btw, the increased uncertainty in the unprocessed scores this year, is also the reason why we found class-global hyper-parameters to generalize better than classwise hyper-parameters. If you were able to increase the stability of the unprocessed scores, however, you may want to consider using classwise hyper params, but make sure to check generalization with an independent dev/test split.

**Comment by frednam93** [comment]
Thanks for sharing your thoughts!
